LLVM_AST/llvm_ast.sml

```diff
@@ -0,0 +1,97 @@
+(*********************************************************************************)
+
+(*                        Project : LLVM bindings for sml                        *)
+(*                        Program : llvm_ast.sml                                 *)
+(*                        Author  : Rahul Dhawan                                 *)
+(*                        Purpose : To capture the AST for LLVM                  *)
+
+(*********************************************************************************)
+
+
+(* Name can be used for the variable and function *)
+
+datatype Name = Name of string
+              | UnName of int 
+              
+(* constants *)
+
+datatype Constant = Int of int
```

Comment:
Convert int to IntInf.int. This is because we want to use multiprecision
integers for constants so that integer constants do not overflow when
the compiler is compiling programs.